Update documentation

kantai opened the pull request:
This is mostly updated by copying content from SIP-002.

I deleted the files that seemed to be getting updated elsewhere (namely in https://github.com/blockstack/docs.blockstack/pull/541) --

* tutorial
* sdk-quickstart

And removed some files that had documentation elsewhere or shouldn't be documented
* install-source -- this is covered in the repo README
* clarityCLI -- this CLI probably shouldn't exist anymore. all of it's features are accesible via `blockstack-cli local`, and there's another clarity cli (javascript), which is more featureful.

I also removed `functions`, since it's content was better covered in the language ref, plus the generated function docs.

kantai commented:
I also maybe need some insight from @hstove or @agraebe on how this documentation relates to https://github.com/blockstack/docs.blockstack/pull/541

There's overlap in these files -- will deleting or editing things here create problems on the documentation site?

jcnelson commented on docs/smart-contract-vm.md:
````diff
@@ -65,7 +65,7 @@ You should see a message saying "Database created".
 
 ```bash
-$ blockstack-core local check sample-programs/names.clar /data/db
+$ blockstack-core local check sample-contracts/names.clar /data/db
````
Is `blockstack-core` still the name of the binary to run?

kantai replied:
Yep -- these commands are also checked in a "demo_tests" job in Circle.

jcnelson commented on docs/smart/clarityRef.md:
```diff
@@ -9,141 +9,126 @@ This file contains the reference for the Clarity language.
+during contract launch. The type system does _not_ have a universal
+super type. The type system contains the following types:
+
+* `(tuple (key-name-0 key-type-0) (key-name-1 key-type-1) ...)` -
```
Do we want to mention implicit tuple syntax?

kantai replied:
I'd prefer not to here -- I want to keep this as close as possible to the SIP and with minimal surface area for now. Once I can setup a better structure for executing automatic tests for any code in the documentation, I'd be happier to expand examples.

jcnelson commented on docs/smart/principals.md:
````diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
-        (credit-balance! recipient amount))
-    true
-    false))
+```cl
````
scheme?

kantai replied:
Fixed in 1a0d04d

njordhov replied:
I have found using triple-tick `clojure` to result in the most fitting formatting.

jcnelson commented on docs/smart/principals.md:
````diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
-    false))
+```cl
+(define-public (transfer-to-recipient! (recipient principal) (amount uint))
+  (if (check-condition)
````
Should `check-condition` be an explicit condition in this example?

kantai replied:
Added in 1a0d04d

jcnelson commented on docs/smart/principals.md:
````diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
 
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```cl
````
scheme?

kantai replied:
Fixed in 1a0d04d

jcnelson commented on docs/smart/principals.md:
````diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
 ```cl
 (define-public (claim-from-faucet)
-  (if (is-none? (fetch-entry claimed-before (tuple (sender tx-sender))))
+  (if (is-none? (fetch-entry claimed-before {sender: tx-sender}))
````
If we're going to use `{ }` in examples, it should be introduced earlier as the way to define tuples.

kantai replied:
There's no documentation for how to declare any literals in these documents at the moment. That could be added as a section to the clarityRef.md file, I think.

kantai replied:
Replaced with the explicit tuple syntax in https://github.com/blockstack/stacks-blockchain/pull/1439/commits/9eb9677a3433ce0d0949dd085ccff96b39976aec

jcnelson commented on docs/smart/principals.md:
```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
             (insert-entry! claimed-before {sender: requester} {claimed: true})
-            (as-contract (stacks-transfer! requester 1)))))
-      (err 1))
+            (as-contract (stacks-transfer? u1 tx-sender requester))))
```
Is it `stacks-transfer?` or `stx-transfer?`

kantai replied:
Fixed in 1a0d04d

kantai commented:
Thanks @jcnelson -- I addressed the comments related to accuracy -- this PR is mostly about fixing the previously inaccurate and out-of-date details in these docs.

These files are still a pretty far way removed from comprising a full language reference, but that will be an issue for another day.

blockstack-devops commented:
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
